src/ibtracert.c

```diff
@@ -811,9 +820,29 @@ int main(int argc, char **argv)
 	argc -= optind;
 	argv += optind;
 
-	if (argc < 2)
+	if (argc < 2 && lid_file[0] == '\0') 
 		ibdiag_show_usage();
 
+	if (argc == 2) {
+		srcid = strdup(argv[0]);
+		dstid = strdup(argv[1]);
+     	}
+
+	if (lid_file != NULL) {
+		lid_fd = fopen(lid_file, "r");
+		if (!lid_fd) {
+			IBEXIT("cannot open lid-file %s", lid_file);
+		} else {
+			srcid = (char*) malloc(6*sizeof(char));
+			dstid = (char*) malloc(6*sizeof(char));
```

Comment:
Check for malloc failures and IBEXIT if so
Also, if GUIDs handled need more than 6 bytes for each malloc

Reply:
will do